wsd/COOLWSD.cpp

```diff
@@ -4167,19 +4167,23 @@ void COOLWSD::processFetchUpdate()
         http::Request request(uriFetch.getPathAndQuery());
         request.add("Accept", "application/json");
 
-        const std::shared_ptr<const http::Response> httpResponse =
-            sessionFetch->syncRequest(request);
-        if (httpResponse->statusLine().statusCode() == http::StatusCode::OK)
-        {
-            LOG_DBG("Infobar update returned: " << httpResponse->getBody());
+        sessionFetch->setFinishedHandler([](const std::shared_ptr<http::Session>& httpSession) {
```

Comment:
`sessionFetch` is local. It will get destroyed when we return. We need a long-lived instance and we need to check if there is a request in flight before we create a new one.

Reply:
@Ashod If we should never create sessions on the stack - can you privatize or protect the constructor, and have a static ::create method that returns a shared_ptr or somesuch ? =) hopefully that would stop others doing this and armor us a bit.

Thanks !

Reply:
@mmeeks, it is precisely as you say: `http::Session` is private and can only be created through a static that returns `shared_ptr`. The issue here was that the `shared_ptr` was a local and would get destroyed once we returned from this function.

Reply:
Hmm - why then does the Response or whatver is holding things there not hold a reference on the session ? - requiring that to be owned by the std::function closure seems not to be a winning strategy ;-)